### DISCUSSION TITLE
Photo Exposure Control

### DISCUSSION THREAD

**Discussion by PixelPincher** [gopro/labs#1194] [Q&A] [answered]
First off: GoPro Labs is a really great initiative: good idea and thank you.

When taking pictures with a GoPro or any camera, I like to set manual exposure settings.
I can use GoPro Labs to set the shutter speed to 1/8 (and almost any faster value: 1/9, 1/10, 1/11...1/4000...),
but I cannot find a way to set the shutter speed to anything between 1/8 and 2 seconds.
That is a really big gap of 4 stops!
no 1/4, 1/2, or 1 second... all values I feel I often need.

**To set shutter speed to 1/8 I used command**:
    mP$EXPX=8$EXPN=8!S
    *(mP = photo mode,  EXPX and EXPN = min and max shutter, !S = take photo)
    *(I use EXPX and EXPN and not just EXPQ, but that's for another post)
    *(Setting a value lower than 8 results in 1/8 or faster if scene is even just a little too bright)
**To set shutter speed to 2 seconds**:
    mPNe2!S
    *(mPN = Night Photo Mode, e2 = 2 seconds, !S = take photo)
**To set shutter speed to 5 seconds**:
    mPNe5!S
    ...
    Trying to set e1, e3, e4, ... any value other than 2,5,10,15,20,30 in "Night Photo Mode" results in "Auto" exposure.

I tried adding
$GCMP=1 (Disable gyro compensated exposure).
But this does not help with forcing 1/8 shutter *(I tried both before and after command mP [set to Photo Mode]).

**So my Questions are**:
Am I missing something?
Are there other ways to control photo shutter speed?
Any way to set an exposure to be 1/2 second or 1 second?
Is this something that require a 'feature request'?
*(Extra: Any ideas how I could force an exposure to be what I set it to be and not have GoPro underexpose the image if the scene is deemed too bright?)

**Comment by dnewman-gpsw** [comment, marked as answer]
Just use Night Photo, is it more flexible.

This works:
mPNeA$EXPQ=3!S

mPNeA$EXPQ=1.5!S

Very flexible. 

EXPQ is the same as using EXPX and EXPM to the same value.

**Reply by PixelPincher** [reply]
Hello dnewman-gpsw and thank you for your response.
This is a great solution and you have solved my query!
But there are some issues / notes I wanted to point out:

[1] mPNeA$EXPQ=0.1!S  sends the camera into a reboot loop. Camera fails to take any photo and glitches out in spectacular ways.  I have to pull out the battery in order to turn off the camera and stop it.
Solution: I will use mPNe10!S instead.

[2] mPNeA$EXPQ=0.2!S  works

**[3]** mPNeA$EXPQ=0.3!S  and  mPNeA$EXPQ=0.4!S  both work to produce good looking images of different exposures, 
but the metadata for both shows the shutter speed to be exactly 3 seconds (both in Adobe Bridge and exiftool.exe).

[4] mPNeA$EXPQ=1!S ,  and with values: 2, 4, 8,... all work great.

**[5]** In all cases, when reviewing the pictures taken with this code on the GoPro camera itself, and clicking [i] for image data, the "Shutter" value reads always and only as "Auto".
It would be nice to have the actual shutter value there.
(When using  mP$EXPX=8$EXPN=8!S  or  mP$EXPQ=8!S  the shutter speed values are shown)

**Reply by dnewman-gpsw** [reply]
[1]  Yes, using 0.1 for tens second would be an illegal operation. Camera is detecting some thing wrong and reboots automatically, this is called a watch dog timer (they are a good thing.) 

Stick to auto exposure $EXPQ=1 or greater.

[3][5] That is beyond Labs.  When feature is polished it becomes main firmware.  Labs is for prototypes are experimentation.
